ops/internal/report/l2.go

```diff
@@ -1,180 +1,91 @@
+		return nil, fmt.Errorf("L2 contracts release must have 'tag://' prefix, got: %s", l2contractsrelease)
 	}
+	tagValue := strings.TrimPrefix(l2contractsrelease, "tag://")
 
-	standardDeployConfig, err := state.CombineDeployConfig(standardIntent, standardIntent.Chains[0], standardState, standardChainState)
-	if err != nil {
-		return standardHash, nil, fmt.Errorf("failed to combine deploy config: %w", err)
-	}
+	var report L2Report
+	report.Release = tagValue
 
-	lgr := log.NewLogger(log.NewTerminalHandlerWithLevel(os.Stderr, log.LevelWarn, false))
-	deployer := common.Address{'D'}
-	host, err := env.DefaultScriptHost(
-		broadcaster.NoopBroadcaster(),
-		lgr,
-		deployer,
-		artifacts,
-	)
+	originalCoreGenesis := &core.Genesis{}
+	originalGenesisByte, err := json.Marshal(originalGenesis)
 	if err != nil {
-		return standardHash, nil, fmt.Errorf("failed to create script host: %w", err)
+		return nil, fmt.Errorf("failed to marshal original genesis: %w", err)
 	}
-
-	if err := opcm.L2Genesis(host, &opcm.L2GenesisInput{
-		L1Deployments: opcm.L1Deployments{
-			L1CrossDomainMessengerProxy: common.Address(*chainCfg.Addresses.L1CrossDomainMessengerProxy),
-			L1StandardBridgeProxy:       common.Address(*chainCfg.Addresses.L1StandardBridgeProxy),
-			L1ERC721BridgeProxy:         common.Address(*chainCfg.Addresses.L1ERC721BridgeProxy),
-		},
-		L2Config: standardDeployConfig.L2InitializationConfig,
-	}); err != nil {
-		return standardHash, nil, fmt.Errorf("failed to call L2Genesis script: %w", err)
+	if err := json.Unmarshal(originalGenesisByte, originalCoreGenesis); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal original genesis: %w", err)
 	}
+	originalGenesisHash := originalCoreGenesis.ToBlock().Hash()
 
-	host.Wipe(deployer)
-
-	standardAllocs, err := host.StateDump()
+	standardCoreGenesis := &core.Genesis{}
+	standardGenesisByte, err := json.Marshal(standardGenesis)
 	if err != nil {
-		return standardHash, nil, fmt.Errorf("failed to dump state: %w", err)
+		return nil, fmt.Errorf("failed to marshal standard genesis: %w", err)
 	}
+	if err := json.Unmarshal(standardGenesisByte, standardCoreGenesis); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal standard genesis: %w", err)
+	}
+	standardGenesisHash := standardCoreGenesis.ToBlock().Hash()
 
-	standardGenesis, err := genesis.BuildL2Genesis(&standardDeployConfig, standardAllocs, &eth.BlockRef{
-		Hash:       startBlock.Hash(),
-		Number:     startBlock.Number.Uint64(),
-		ParentHash: startBlock.ParentHash,
-		Time:       startBlock.Time,
-	})
-	if err != nil {
-		return standardHash, nil, fmt.Errorf("failed to build standard genesis: %w", err)
+	if originalGenesisHash != standardGenesisHash {
+		return nil, fmt.Errorf("original genesis hash does not match standard genesis hash")
 	}
 
-	diffs := DiffAllocs(standardAllocs.Accounts, originalAllocs)
-	return standardGenesis.ToBlock().Hash(), diffs, nil
+	report.ProvidedGenesisHash = originalGenesisHash
```

Comment:
Let's diff these using `OpaqueMapping` since Geth's `core.Genesis` struct changes between hardforks.

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1049/commits/21c3eb83b8b7d1105a570327102fb52fb07240ac